README.md

```diff
@@ -260,6 +260,78 @@ Now, any script you defined in the ScriptsFunction will be added to the HTML tog
 
 > Tip: You could use it together with useShouldHydrate to disable Remix scripts in certain routes but still load scripts for analytics or small features that need JS but don't need the full app JS to be enabled.
 
+
+### StructuredData
+
+If you need to include structured data (JSON-LD) scripts on certain routes, you can use the `StructuredData` component together with the `HandleStructuredData` type or `StructuredDataFunction` type.
+
+In the route you want to include the structured data, add a `handle` export with a `structuredData` method, this method should implement the `StructuredDataFunction` type.
+
+```ts
+import type { WithContext, BlogPosting } from 'schema-dts';
+
+// export the handle with the correct type:
+export let handle: HandleStructuredData<LoaderData> = {
```

Comment:
What do you think if you pass a second generic to `HandleStructuredData` where you can send `BlogPosting`? And use that to ensure the return type of `StructuredDataFunction` is a `WithContext<BlogPosting>` type. That way you can return directly inside the function without assigning to a variable.

A thing to consider is what happens when there's no data, the return type of `StructuredDataFunction` could be `WithContext<Result> | null` and return `null` when you can't return anything.

If you want to return an array instead of an object I suppose you could do `HandleStructuredData<LoaderData, BlogPosting[]>` and that should work?

Reply:
`null` is a simple case to support.

I like the idea of making the `Thing` type explicit, if you want. The single element is easy to type, but I'll have to work with supporting the array well. My use-case of `BreadcrumbList` + `BlogPosting` would need to specify two types with the resulting type being `[WithContext<BreadcrumbList>, WithContext<BlogPosting>]`.